core/auth/payment_metadata.go

```diff
@@ -0,0 +1,168 @@
+package auth
+
+import (
+	"crypto/ecdsa"
+	"fmt"
+	"math/big"
+
+	"github.com/Layr-Labs/eigenda/core"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/math"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/signer/core/apitypes"
+)
+
+/* SUBJECT TO MODIFICATIONS */
+
+// EIP712Signer handles EIP-712 domain specific signing operations over typed and structured data
+type EIP712Signer struct {
+	domain apitypes.TypedDataDomain
+	types  apitypes.Types
+}
+
+// NewEIP712Signer creates a new EIP712Signer instance
+func NewEIP712Signer(chainID *big.Int, verifyingContract common.Address) *EIP712Signer {
+	return &EIP712Signer{
+		domain: apitypes.TypedDataDomain{
+			Name:              "EigenDA",
+			Version:           "1",
+			ChainId:           (*math.HexOrDecimal256)(chainID),
+			VerifyingContract: verifyingContract.Hex(),
+		},
+		types: apitypes.Types{
+			"EIP712Domain": []apitypes.Type{
+				{Name: "name", Type: "string"},
+				{Name: "version", Type: "string"},
+				{Name: "chainId", Type: "uint256"},
+				{Name: "verifyingContract", Type: "address"},
+			},
+			"PaymentMetadata": []apitypes.Type{
+				{Name: "accountID", Type: "string"},
+				{Name: "binIndex", Type: "uint32"},
+				{Name: "cumulativePayment", Type: "uint64"},
+				{Name: "dataLength", Type: "uint32"},
```

Comment:
I'm not sure the dataLength would be needed in v2 if we also have the blob key.

Reply:
Could I assume meterer will receive validated blob header which contains payment metadata and the quorum numbers and length? If so I could remove `validateSignature` from the meterer process and make a new blobHeader struct?